CollapseLauncher/Program.cs

```diff
@@ -229,6 +234,15 @@ private static async Task InitDatabaseHandler()
             await DbHandler.Init();
         }
 
+        public static void SystemEvent_EndingSession(object sender, SessionEndingEventArgs e)
+        {
+            if (!IsCriticalOpInProgress) return;
+
+            e.Cancel = true;
+            ShutdownBlocker.StartBlocking(WindowUtility.CurrentWindowPtr, Lang._Dialogs.EnsureExitSubtitle,
+                                          ILoggerHelper.GetILogger("ShutdownBlocker"));
+        }
+
```

Comment:
Same as above. We ca remove this method as it's the same functionality as WndProc's ``WM_QUERYENDSESSION``